provider/provider.go

```diff
@@ -104,14 +104,23 @@ func (a *azureProvider) CreateInstance(ctx context.Context, bootstrapParams para
 		}
 	}()
 
-	_, err = a.azCli.CreateVirtualNetwork(ctx, runnerSpec.BootstrapParams.Name, runnerSpec.VirtualNetworkCIDR)
-	if err != nil {
-		return params.ProviderInstance{}, fmt.Errorf("failed to create virtual network: %w", err)
-	}
+	var subnetID string
 
-	subnet, err := a.azCli.CreateSubnet(ctx, runnerSpec.BootstrapParams.Name, runnerSpec.VirtualNetworkCIDR)
-	if err != nil {
-		return params.ProviderInstance{}, fmt.Errorf("failed to create subnet: %w", err)
+	if runnerSpec.VnetSubnetID != "" {
```

Comment:
I think we can split this in a separate helper function that takes the spec and the context as an argument and returns the subnet id and an error. Something like

```go
func (a *azureProvider) resolveSubnetID(ctx context.Context, spec spec.RunnerSpec) (string, error) {
     // does the thing
    return subnetID, nil
}
```